lib/rules/require-jsdoc.js

```diff
@@ -93,8 +83,11 @@ module.exports = {
                 }
             },
             FunctionExpression(node) {
-                if (options.MethodDefinition) {
-                    checkClassMethodJsDoc(node);
+                if (
+                    (options.MethodDefinition && node.parent.type === "MethodDefinition") ||
+                    (options.FunctionExpression && (node.parent.type === "VariableDeclarator" || node.parent.type === "Property"))
```

Comment:
Should the `node.parent.type === "Property"` check also verify that `node === node.parent.value`?

Otherwise, I think something like this will be reported:

```js
var foo = {
    [function() {}]: 1
};
```

Reply:
Good catch!